packages/@react-aria/combobox/docs/useComboBox.mdx

```diff
@@ -499,7 +497,7 @@ function ControlledComboBox() {
       isOpen: true,
       inputValue: value,
       selectedKey: value === '' ? null : prevState.selectedKey,
-      items: filteredItems
+      items: optionList.filter(item => startsWith(item.name, value))
```

Comment:
this was broken because the memoized filteredItems is one step behind the user input (aka the latest input value is set after this setState but the `items` set here were based on the previous input value). This should fix